bridges/BakaUpdatesMangaReleasesBridge.php

```diff
@@ -0,0 +1,84 @@
+<?php
+class BakaUpdatesMangaReleasesBridge extends BridgeAbstract {
+	const NAME = 'Baka Updates Manga Releases';
+	const URI = 'https://www.mangaupdates.com/';
+	const DESCRIPTION = 'Get the latest series releases';
+	const MAINTAINER = 'fulmeek';
+	const PARAMETERS = array(array(
+		'series_id' => array(
+			'name'			=> 'Series ID',
+			'exampleValue'	=> '12345'
+		)
+	));
+	const LIMIT_ITEMS = 10;
+
+	private $feedName = '';
+
+	public function collectData() {
+		$series_id = preg_replace('/[^0-9]/', '', $this->getInput('series_id'));
+		if (empty($series_id))
+			returnServerError('Invalid series ID');
+
+		$html = getSimpleHTMLDOM(self::URI . 'releases.html?search=' . $series_id . '&stype=series')
+			or returnServerError('Series not found');
+
+		$objTitle = $html->find('td[class="text pad"]', 1);
+		if ($objTitle)
+			$this->feedName = $objTitle->plaintext;
+
+		$itemlist = $html->find('td#main_content table table table tr');
+		if (!$itemlist)
+			returnServerError('No releases');
+
+		$limit = self::LIMIT_ITEMS;
+		foreach($itemlist as $element) {
+			if (!$element->find('td[class="text pad"]', 0))
```

Comment:
This operation is repeated six times in total, which is very inefficient. Instead, assign the result to a parameter and use it like an array instead:

```suggestion
			$cols = $element->find('td[class="text pad"]');
			if (!$cols)
```

Then later just access each column by using `$cols[n]`, where `n` is the column number.

Reply:
No, since there is no way to query only the rows that has matching tds in there, I have to fetch them all and test the first cell. This appears to be a limitation of simplehtmldom since it does not support selectors like ":has(...)".
I wanted to fetch rows that match 'td#main_content table table table tr:has(td[class="text pad"])', but that's not possible. I'm open for any suggestions.

Reply:
Sorry, my suggestion above might be misleading.

The conditions don't change, just the way each element is accessed. The way `find` works, it always returns an array of matching elements. In this case, `find('td[class="text pad"]');` returns an array of `td` elements whose class is exactly `text pad`. There are two possible results:

1) An empty array (no match found)
2) An array of matching `td` elements (title, date, etc...)

Now each element can be checked like before, but it is not necessary to call `find` again (which is slow). Here is a complete example of what I mean. Functionally, there is no difference to the current implementation.

<details><summary>Show Example</summary>

```PHP
	public function collectData() {
		$html = getSimpleHTMLDOM($this->getURI())
			or returnServerError('Series not found');

		$objTitle = $html->find('td[class="text pad"]', 1);
		if ($objTitle)
			$this->feedName = $objTitle->plaintext;

		$itemlist = $html->find('td#main_content table table table tr');
		if (!$itemlist)
			returnServerError('No releases');

		$limit = self::LIMIT_ITEMS;
		foreach($itemlist as $element) {
			$cols = $element->find('td[class="text pad"]');
			if (!$cols)
				continue;
			if ($limit <= 0)
				break;

			$item = array();
			$title = array();

			$item['content'] = '';

			$objDate = $cols[0];
			if ($objDate)
				$item['timestamp'] = strtotime($objDate->plaintext);

			$objTitle = $cols[1];
			if ($objTitle) {
				$title[] = html_entity_decode($objTitle->plaintext);
				$item['content'] .= '<p>Series: ' . $objTitle->innertext . '</p>';
			}

			$objVolume = $cols[2];
			if ($objVolume && !empty($objVolume->plaintext))
				$title[] = 'Vol.' . $objVolume->plaintext;

			$objChapter = $cols[3];
			if ($objChapter && !empty($objChapter->plaintext))
				$title[] = 'Chp.' . $objChapter->plaintext;

			$objAuthor = $cols[4];
			if ($objAuthor && !empty($objAuthor->plaintext)) {
				$item['author'] = html_entity_decode($objAuthor->plaintext);
				$item['content'] .= '<p>Groups: ' . $objAuthor->innertext . '</p>';
			}

			$item['title']	= implode(' ', $title);
			$item['uri']	= $this->getURI() . '#' . hash('sha1', $item['title']);

			$this->items[] = $item;

			--$limit;
		}
	}
```

</details><br>

Reply:
Now I see what you mean. Of course, you're right. Thanks :)